### PR TITLE
Refactor panoptic quality

### PR DESCRIPTION
**Pull request by marcocaccin**
## What does this PR do?

Preparation work for  #1591: breaks apart the big messy update() function, and moves all the helper functions to a new `src/torchmetrics/functional/detection/_panoptic_quality_common.py` file.

To make the review a lot simpler, the PR contains 2 commits: I suggest to look at the first commit 7cbf54b6743aa251a3e69ae801788dc958aebe60 which contains all the new code and it can be `diff`'d against `master`, whereas the 2nd commit d873aabc3140f390b4eaef2fe4180c7b1734c8e0 is doing nothing else than moving code around.

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by marcocaccin** [comment]
Not sure how to interpret the codecov failure, the PR doesn't add any "new" code. Is it a CI glitch? Should I do something about it?

File: src/torchmetrics/detection/panoptic_quality.py [hunk 1]
```diff
@@ -11,13 +11,12 @@
...
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import warnings
 from typing import Any, Collection, Optional, Sequence, Union
 
 import torch
 from torch import Tensor
 
-from torchmetrics.functional.detection.panoptic_quality import (
+from torchmetrics.functional.detection._panoptic_quality_common import (
```
Comment (Borda) [inline src/torchmetrics/detection/panoptic_quality.py:19]: What is the motivation for moving it to a separate file when common still means shared only for this one metric? it generates a much larger diff so it is not so easy to track real changes... :chipmunk:

**Comment by Borda** [comment]
> I suggest to look at the first commit [7cbf54b](https://github.com/Lightning-AI/metrics/commit/7cbf54b6743aa251a3e69ae801788dc958aebe60) which contains all the new code and it can be `diff`'d against `master`

let's just stay with this one commit pls :otter:

**Comment by Borda** [comment]
> Not sure how to interpret the codecov failure, the PR doesn't add any "new" code. Is it a CI glitch? Should I do something about it?

it is a sanity check that the new code is covered, but it also has some false negative alarms when the coverage change is almost zero...

File: src/torchmetrics/detection/panoptic_quality.py (hunk 1 above)
Comment (marcocaccin) [reply src/torchmetrics/detection/panoptic_quality.py:19]: The common functionality is being pulled out into a separate file according to what I mentioned in #1591 . This PR is the first of 2, so that you don’t get hit with an unreviewable one… if there’s a better option let me know!

Comment (Borda) [reply src/torchmetrics/detection/panoptic_quality.py:19]: I understand common, but while it is used only by one metric it seems rather specific than common :)

Comment (marcocaccin) [reply src/torchmetrics/detection/panoptic_quality.py:19]: Should I call it utils in this PR and rename it to common in the next for consistency? Alternatively all of the common code can stay in panoptic_quality.py and the modified implementation will import from there, but the asymmetry would be slightly confusing

Comment (Borda) [reply src/torchmetrics/detection/panoptic_quality.py:19]: it is not about the name, but what other function/metric would use it?

Comment (marcocaccin) [reply src/torchmetrics/detection/panoptic_quality.py:19]: The modified PQ metric that is mentioned in the issue and I’m implementing on the side of the standard PQ that’s already there

Comment (Borda) [reply src/torchmetrics/detection/panoptic_quality.py:19]: ok, I see now, sorry :innocent:

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/detection/panoptic_quality.py (hunk 1 above)
Comment (marcocaccin) [reply src/torchmetrics/detection/panoptic_quality.py:19]: No worries, it was clear we had a misunderstanding but I wasn't sure where 🙂

Comment (marcocaccin) [reply src/torchmetrics/detection/panoptic_quality.py:19]: Btw you had a very good reason to be confused, I am now too... https://github.com/Lightning-AI/metrics/issues/1591#issuecomment-1456985706. Let me know if you have a preference either way, we don't need to merge this PR if there's a preference in keeping just 1 metric with 2 flavours.
